### PR TITLE
Implement matchers refactoring

### PR DESCRIPTION
**Pull request by jwaimann**
@patricioe @dendril @mravinale 

Now sdk supports attributes of type string, datetime and long

### CODE REVIEW

File: NetSDK/Services/Parsing/Classes/CompareMatcher.cs [hunk 1]
```diff
@@ -3,29 +3,33 @@
...
         protected long start;
         protected long end;
-
         public bool Match(string key)
         {
             switch (dataType)
             {
                 case DataTypeEnum.DATETIME:
-                    return MatchDate(key);
+                    var date = key.ToDateTime();
```
Comment (dendril) [inline NetSDK/Services/Parsing/Classes/CompareMatcher.cs]: What happen if key is "-1" and the Matcher Type is DATETIME?

File: NetSDK/Services/Parsing/Classes/CompareMatcher.cs [hunk 2]
```diff
@@ -3,29 +3,33 @@
...
+                    return date != null ? Match(date.Value) : false;
                 case DataTypeEnum.NUMBER:
-                    return MatchNumber(key);
+                    long number;
+                    var result = long.TryParse(key, out number);
```
Comment (dendril) [inline NetSDK/Services/Parsing/Classes/CompareMatcher.cs]: What happen if key is a letter instead of a number?

Comment (jwaimann) [inline NetSDK/Services/Parsing/Classes/CompareMatcher.cs]: Covered in unit tests : https://github.com/splitio/.net-client/blob/65aa2a6c70e045bd59e36cd9c1022176c2814a8f/Net%20SDK%20Unit%20Tests/Unit%20Tests/Matchers/EqualToMatcherTests.cs#L34

File: NetSDK/Services/Parsing/Classes/CompareMatcher.cs (hunk 1 above)
Comment (jwaimann) [inline NetSDK/Services/Parsing/Classes/CompareMatcher.cs]: Covered in unit tests: https://github.com/splitio/.net-client/blob/65aa2a6c70e045bd59e36cd9c1022176c2814a8f/Net%20SDK%20Unit%20Tests/Unit%20Tests/Matchers/EqualToMatcherTests.cs#L62

**Pull request merged**